src/trio/_core/_run.py

```diff
@@ -468,10 +486,22 @@ def recalculate(self) -> None:
                 or current.parent_cancellation_is_visible_to_us
             )
             if new_state != current.effectively_cancelled:
+                if (
+                    current._scope._cancel_reason is None
+                    and current.parent_cancellation_is_visible_to_us
+                ):
+                    assert current._parent is not None
+                    current._scope._cancel_reason = (
+                        current._parent._scope._cancel_reason
+                    )
                 current.effectively_cancelled = new_state
                 if new_state:
                     for task in current._tasks:
                         task._attempt_delivery_of_any_pending_cancel()
+                    # current._cancel_reason will not be None, unless there's misnesting
+                    for child in current._children:
+                        if child._scope._cancel_reason is None:
+                            child._scope._cancel_reason = current._scope._cancel_reason
```

Comment:
Why do we both propagate down here and then copy the parent's reason above? Can't we just do either of these?

Reply:
Oh you are very right. Deleting the former broke tests, but deleting the latter did not.